benchmark/cpp_xorUpdateRandom_readRandom.sh

```diff
@@ -0,0 +1,257 @@
+me=`basename $0`
+if [ $# -eq 0 ]; then
+  echo "ERROR: No db path provided."
+  echo "Usage: ${me} <path-to-db>"
+  exit 1
+fi
+
+############################################
+####   General Benchmark                ####
+############################################
+# benchmarks  -- see each benchmark
+
+num=$((1 * 1000 * 1000))
+reads=$num
+key_size=2
+value_size=$((4 * 1024))
+
+# Number of concurrent threads to run for the benchmark.
+threads=2
+
+
+############################################
+####   Statistics                       ####
+############################################
+# Report database statistics (true/false)
+statistics=1
+
+# Print histogram of operation timings (true/false)
+histogram=1
+
+# a temp varibale that estimates the total size of an entry
+entry_size=$(($key_size + $value_size))
+
+# Stats are reported every N operations when this is greater than zero.
+# When 0 the interval grows over time. The formula below comes out of an
+# experiment where we had an 100K interval with 16B keys and 8B values.
+# interval*entry_size=100K*(16+8), hence interval = 2400K/entry_size
+stats_interval=$((240000000 / $entry_size))
```

Comment:
I have done a lot of work to make tools/run_flash_bench.sh configurable. It
runs many tests via tools/benchmark.sh and now via environment variables
you can set things like memory (write_buffer_size), number of background
threads and number of "user" threads. Without looking at your diff again it
might be good to do that here. However, if you statically configure this
for a small server (low memory, low threads) then that should be fine.

On Wed, Apr 15, 2015 at 5:12 AM, Pooya Shareghi <notifications@github.com>
wrote:

> In benchmark/cpp_xorUpdateRandom_readRandom.sh
> <https://github.com/facebook/rocksdb/pull/575#discussion_r28365462>:
>
> > +####   Statistics                       ####
> > +############################################
> > +# Report database statistics (true/false)
> > +statistics=1
> > +
> > +# Print histogram of operation timings (true/false)
> > +histogram=1
> > +
> > +# a temp varibale that estimates the total size of an entry
> > +entry_size=$(($key_size + $value_size))
> > +
> > +# Stats are reported every N operations when this is greater than zero.
> > +# When 0 the interval grows over time. The formula below comes out of an
> > +# experiment where we had an 100K interval with 16B keys and 8B values.
> > +# interval*entry_size=100K*(16+8), hence interval = 2400K/entry_size
> > +stats_interval=$((240000000 / $entry_size))
>
> Great work! I will add that as well. Since stats_interval_seconds
> overrides stats_interval, I will keep both.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/rocksdb/pull/575/files#r28365462>.
>



-- 
Mark Callaghan
mdcallag@gmail.com